content/docs/work-with-source-git.md

```diff
@@ -0,0 +1,123 @@
+---
+title: "Source-git"
+date: 2020-06-25T23:59:59+01:00
+draft: false
+chapter: false
+weight: 2
+aliases:
+- /source-git
+- /omfg
+---
+
+This document is focused on how packit treats a source-git repo.
+
+In short, source-git concept is fairly simple - take an upstream repo and shove
+downstream content on top of it. Easy right?
+
+Packit is then able to work with such a repo and:
+
+* Create SRPMs
+
+* Sync the state into dist-git
+
+* And you are able to utilize all features of packit-service: builds on PRs,
+  tests on those builds, update your package in Fedora.
+
+Let's describe all of these in detail.
+
+
+### Upstream history or not
+
+Your source-git repo can contain upstream git history if you want. Or it
+doesn't. This is really up to you and packit is able to work with both.
+
+The key ingredient here is the config option `upstream_ref`. It tells packit
+the git ref which is suppose to match the archive specified in the spec file.
+All commits on top of this ref are treated either as:
+
+* downstream patches
+* downstream packaging
+
+For downstream patches, packit creates a patch from git commits on top of
+`upstream_ref`. Downstream packaging changes are ignored in patch files since
+they can be directly committed to dist-git. You can control the ignore
+mechanism with config option `patch_generation_ignore_paths`.
+
+
+#### Example
+
+Let's have a look at source-git repo for pacemaker package:
+```
+$ git log --online
+bd12722 (HEAD -> c8s-test, origin/c8s-test, master) downstream packaging
+7201e28 Refactor: controller: remove unused argument
+91c557c Refactor: controller: convert active_op_t booleans to bitmask
+2e90063 Refactor: controller: rename struct recurring_op_s to active_op_t
+3ee90ce (tag: source-git-start) pacemaker-2.0.3 base
+```
+
+We have 5 commits:
+* 3ee90ce - the bottom commit, equals to an unpackaed upstream tarball
+* 2e90063 - a downstream patch
+* 7201e28 - a downstream patch
+* 91c557c - a downstream patch
+* bd12722 - top commit, contains a spec file so packit won't create a patch file out of this one
```

Comment:
Probably would be more easy to read if these explanations would be in the same order as the commit list above.